lib/string_reverse.rb

```diff
@@ -1,4 +1,24 @@
 # A method to reverse a string in place.
 def string_reverse(my_string)
-  my_string << "not implemented"
+  if my_string == nil
+    return nil
+  elsif my_string == ''
+    return ''
+  elsif my_string.length == 1
+    return my_string
+  end
+
+  beg_index = 0
+  end_index = my_string.length - 1
+  temp = 0
+
+  while beg_index < end_index
+    temp = my_string[beg_index]
+    my_string[beg_index] = my_string[end_index]
+    my_string[end_index] = temp
+    beg_index += 1
+    end_index -= 1
+  end
+
+  return my_string

```

Comment:
Minor: Since this method reverses the string in-place, you don't need to return my_string and lines 4, line 6, line 8 and line 23 could all simply be `return`.